scheduler/mesos/mesos.go

```diff
@@ -0,0 +1,88 @@
+package mesos
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/discovery"
+	"github.com/docker/swarm/scheduler/options"
+	"github.com/samalba/dockerclient"
+)
+
+var ErrNotImplemented = errors.New("not implemented in the mesos scheduler")
+
+type MesosScheduler struct {
+	sync.Mutex
+
+	//TODO: list of mesos masters
+	cluster *cluster.Cluster
+	options *options.SchedulerOptions
+}
+
+func (s *MesosScheduler) Initialize(options *options.SchedulerOptions) {
+	s.options = options
+
+	s.cluster = cluster.NewCluster(s.options.Store)
+}
+
+// Schedule a brand new container into the cluster.
+func (s *MesosScheduler) CreateContainer(config *dockerclient.ContainerConfig, name string) (*cluster.Container, error) {
+
+	s.Lock()
+	defer s.Unlock()
+
+	//TODO: RequestOffers from mesos master
+
+	//TODO: pick the right offer (using strategy & filters ???)
+
+	//TODO: LaunchTask on the Mesos cluster and get container
+
+	//TODO: Store container in store
+	//  -   s.cluster.CommitContainerInStore(container.Id, config, name)
```

Comment:
Because when you do a `docker run` with the cli, it is in fact 3 http calls: `create` + `start` + `attach`

So I imagined we could do the `create` through mesos and the `start` + `attach` through the direct connection we have to the docker node.

Also, our goal, is to have 100% of the CLI work in swarm, either with mesos or the builtin.
We need this to support `docker create`

I know mesos is calling the docker cli, are you calling `docker run` or `docker create` + `docker start` ?